Sources/DistributedSystem/DistributedSystem.swift

```diff
@@ -958,25 +956,25 @@ public class DistributedSystem: DistributedActorSystem, @unchecked Sendable {
             guard let actorInfo = self.actors[actor.id] else {
                 throw DistributedSystemErrors.noConnectionForActor(actor.id)
             }
-            let (channel, suspended) = switch actorInfo {
+            let suspended = switch actorInfo {
             case let .remoteClient(rcs), let .remoteService(rcs):
-                (rcs.channel, rcs.suspended)
+                rcs.suspended
             default:
-                fatalError("Internal error: invalid actor state \(actor.id): \(actorInfo)")
+                fatalError("internal error: invalid actor state \(actor.id): \(actorInfo)")
+            }
+
+            let channelID = actor.id.channelID
+            guard var channelInfo = self.channels[channelID] else {
+                fatalError("internal error: channel \(channelID) not registered")
             }
 
             if callID != 0 {
-                let ptr = Self.ptr(for: channel)
-                if var channelInfo = self.channels[ptr] {
-                    channelInfo.pendingSyncCalls.insert(callID)
-                    self.channels[ptr] = channelInfo
-                    logger.trace("add sync call \(callID) for \(ptr)")
-                } else {
-                    logger.error("internal error: channel \(ptr) not registered")
-                }
+                channelInfo.pendingSyncCalls.insert(callID)
+                self.channels[channelID] = channelInfo
```

Comment:
BTW, it's possible to update a dictionary element "in-place".

Reply:
I know, but better to keep code simpler from my point of view.
Hope compiler will do it properly :)